app/uk/gov/hmrc/eusubsidycompliancefrontend/models/DateFormValues.scala

```diff
@@ -0,0 +1,120 @@
+
+import play.api.data.Forms.{text, tuple}
+import play.api.data.Mapping
+import play.api.libs.json.Json
+
+import java.time.{LocalDate, ZoneId}
+import java.time.format.DateTimeFormatter
+import scala.util.Try
+
+case class DateFormValues(day: String, month: String, year: String) {
+  def isValidDate: Boolean = try {
+    val dateText = s"${"%02d".format(day.toInt)}/${"%02d".format(month.toInt)}/$year"
+    LocalDate.parse(dateText, DateTimeFormatter.ofPattern("dd/MM/yyyy"))
+    true
+  }
+  catch {
+    case _ => false
+  }
+
+}
+
+case object DateFormValues {
+
+  def trim(inputStr: String) = inputStr.trim()
+
+  lazy val vatRegDateMapping: Mapping[DateFormValues] = tuple(
+    "day"   -> text,
+    "month" -> text,
+    "year"  -> text
+  ).verifying(
+    "error.date.emptyfields",
+    x =>
+      x match {
+        case (d: String, m: String, y: String) if trim(d) == "" && trim(m) == "" && trim(y) == "" => false
+        case _                                                                                      => true
+      })
+    .verifying(
+      "error.day.missing",
+      x =>
+        x match {
+          case (d: String, m: String, y: String) if trim(d) == "" && trim(m) != "" && trim(y) != "" => false
+          case _                                                                                      => true
+        })
+    .verifying(
+      "error.month.missing",
+      x =>
+        x match {
+          case (d: String, m: String, y: String) if trim(d) != "" && trim(m) == "" && trim(y) != "" => false
+          case _                                                                                      => true
+        })
+    .verifying(
+      "error.year.missing",
+      x =>
+        x match {
+          case (d: String, m: String, y: String) if trim(d) != "" && trim(m) != "" && trim(y) == "" => false
+          case _                                                                                      => true
+        })
+    .verifying(
+      "error.day-and-month.missing",
+      x =>
+        x match {
+          case (d: String, m: String, y: String) if trim(d) == "" && trim(m) == "" && trim(y) != "" => false
+          case _                                                                                      => true
+        })
+    .verifying(
+      "error.month-and-year.missing",
+      x =>
+        x match {
+          case (d: String, m: String, y: String) if trim(d) != "" && trim(m) == "" && trim(y) == "" => false
+          case _                                                                                      => true
+        })
+    .verifying(
+      "error.day-and-year.missing",
+      x =>
+        x match {
+          case (d: String, m: String, y: String) if trim(d) == "" && trim(m) != "" && trim(y) == "" => false
+          case _                                                                                      => true
+        })
+    .verifying(
+      "error.date.invalid",
```

Comment:
I think we can put a check here that number of characters in year shouldn't be < 4.  if you put 0 in year , it's a valid year

Reply:
sure 👍